x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/util/installation.ts

```diff
@@ -31,105 +30,100 @@ const installPrebuiltRules = async (
   savedObjectsClient: SavedObjectsClientContract,
   detectionRulesClient: IDetectionRulesClient
 ): Promise<UpdateRuleMigrationInput[]> => {
-  const ruleAssetsClient = createPrebuiltRuleAssetsClient(savedObjectsClient);
-  const ruleObjectsClient = createPrebuiltRuleObjectsClient(rulesClient);
-  const ruleVersionsMap = await fetchRuleVersionsTriad({
-    ruleAssetsClient,
-    ruleObjectsClient,
-  });
-  const { currentRules, installableRules } = getRuleGroups(ruleVersionsMap);
-
-  const rulesToUpdate: UpdateRuleMigrationInput[] = [];
-  const assetsToInstall: PrebuiltRuleAsset[] = [];
-  rulesToInstall.forEach((ruleToInstall) => {
-    // If prebuilt rule has already been installed, then just update migration rule with the installed rule id
-    const installedRule = currentRules.find(
-      (rule) => rule.rule_id === ruleToInstall.elastic_rule?.prebuilt_rule_id
-    );
-    if (installedRule) {
-      rulesToUpdate.push({
-        id: ruleToInstall.id,
-        elastic_rule: {
-          id: installedRule.id,
-        },
-      });
-      return;
-    }
+  // Get required prebuilt rules
+  const prebuiltRulesIds = rulesToInstall
+    .flatMap((rule) => rule.elastic_rule?.prebuilt_rule_id ?? [])
+    .filter((value, index, self) => self.indexOf(value) === index);
```

Comment:
It seems we can extract a `getUniquePrebuiltRuleIds(migrationRules)` function or something so we don't repeat code.

Reply:
done